bugsnag-android-core/src/main/java/com/bugsnag/android/internal/DocumentPath.kt

```diff
@@ -199,28 +219,40 @@ class DocumentPath(path: String) {
             return buff.toString()
         }
 
-        /**
-         * Check if the path string ends in a path separator '.'
-         *
-         * Note: This is 40% faster than using regex.
-         */
-        private fun doesPathEndWithPathSeparator(path: String): Boolean {
-            if (path[path.length - 1] != PATH_SEPARATOR) {
-                return false
+        private fun getPathType(path: String): PathType {
+            if (path.length == 0) {
+                return PathType.PATH_TYPE_NORMAL
             }
-            if (path.length == 1) {
-                return true
+
+            val guessedPathType = when (path[path.lastIndex]) {
+                PATH_SEPARATOR -> PathType.PATH_TYPE_LIST_INSERT
+                ADD_OPERATOR -> PathType.PATH_TYPE_ADD
+                else -> PathType.PATH_TYPE_NORMAL
             }
-            var isUnescapedDot = true
-            for (i in path.length - 2 downTo 0) {
-                val ch = path[i]
-                isUnescapedDot = if (ch == ESCAPE_CHAR) {
-                    !isUnescapedDot
-                } else {
-                    break
+
+            if (path.length > 1 && path[path.lastIndex - 1] == ESCAPE_CHAR) {
+                return PathType.PATH_TYPE_NORMAL
+            }
```

Comment:
it looks like this case should be earlier, since it does not require `guessedPathType` and can be ruled out based solely on `path`.